### PR TITLE
Slack Error Handling

### PR DESCRIPTION
**Pull request by atoff**
Closes CORE-90

### CODE REVIEW

**Comment by atoff** [comment]
@CalumTowers can you check that this is the kind of thing you are after?

**Comment by CalumTowers** [comment]
@atoff Yeah this is where we need to go. I want to try and eliminate the "its Slack's fault" from our BugSnag to...

a) tidy it up
b) stop using our quota

I mean, we do need to optimise our calls to Slack's API to prevent us being blocked/timed out but if its a temporary error with no real impact to the user, I don't want BugSnag (and my phone...) going crazy

File: app/Console/Commands/Command.php [hunk 1]
```diff
@@ -206,4 +216,35 @@ protected function getAuthorLink()
             return 'https://gitlab.com/vatsim-uk/core/blob/development'.$directory;
         }
     }
+
+    private function handleSlackException($e)
+    {
+        $errorClass = get_class($e);
+        $errorCode = $e->getCode();
+        if ($errorClass == ClientException::class) {
```
Comment (CalumTowers) [inline app/Console/Commands/Command.php]: I may be being a little thick here but what is the purpose of checking the error class? Can we not just grab the code, if 408 or 504, break else BugSnag?

Comment (atoff) [reply app/Console/Commands/Command.php]: Good point. I will make it catch the `GuzzleHttp\Exception\BadResponseException` which will catch both.

**Pull request merged**